### PR TITLE
feat: Value streaming respects io manipulators

### PR DESCRIPTION
**Pull request by devjgm**
This PR changes the `Value` `op<<` operator for the types `bool`,
`int64_t`, and `double` so that they print by default as they would as
normal C++ types. This also enables them to work with C++ IO stream
manipulators. For example, this change makes the following work:

```cc
std::cout << std::hex << Value(42);  // Output: 2a
```

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1362)

### CODE REVIEW

File: google/cloud/spanner/value_test.cc [hunk 1]
```diff
@@ -937,8 +938,17 @@ TEST(Value, CommitTimestamp) {
...
+//
+// 1. The test case's name. This is for gtest and needs to be unique.
+// 2. The `Value` instance to stringify.
+// 3. The expect stringified string value
+// 4. An optional lambda (nullptr is allowed) that may apply IO manipulators to
+//    the ostream that may affect the rendered output of the stringified `Value.
 class ValueStreamOperator
-    : public TestWithParam<std::tuple<std::string, Value, std::string>> {
+    : public TestWithParam<std::tuple<std::string, Value, std::string,
+                                      std::function<void(std::ostream&)>>> {
```
Comment (devbww) [inline google/cloud/spanner/value_test.cc]: Might it be nicer to make this non-nullable and change to `std::function<std::ostream(std::ostream&)>`?

File: google/cloud/spanner/value.cc [hunk 2]
```diff
@@ -144,10 +143,8 @@ std::ostream& StreamHelper(std::ostream& os, google::protobuf::Value const& v,
       return os << ')';
     }
 
-    case google::spanner::v1::TYPE_CODE_UNSPECIFIED:
-    case google::spanner::v1::TypeCode_INT_MIN_SENTINEL_DO_NOT_USE_:
-    case google::spanner::v1::TypeCode_INT_MAX_SENTINEL_DO_NOT_USE_:
-      break;
+    default:
```
[the remark this answers is not in the document]
Comment (devjgm) [reply google/cloud/spanner/value.cc]: Possibly. But I don't think we should be mentioning proto fields w/ "DO_NOT_USE" in the name.

File: google/cloud/spanner/value_test.cc (hunk 1 above)
Comment (devjgm) [reply google/cloud/spanner/value_test.cc]: Done.

File: google/cloud/spanner/value_test.cc [hunk 3]
```diff
@@ -937,101 +936,124 @@ TEST(Value, CommitTimestamp) {
...
+  };
+  auto const boolalpha = [](std::ostream& os) -> std::ostream& {
+    return os << std::boolalpha;
+  };
+  auto const float4 = [](std::ostream& os) -> std::ostream& {
+    return os << std::showpoint << std::setprecision(4);
+  };
+  auto const alphahex = [](std::ostream& os) -> std::ostream& {
+    return os << std::boolalpha << std::hex;
+  };
```
Comment (devbww) [inline google/cloud/spanner/value_test.cc]: Possibly
```
  auto const alphahex = [&hex, &boolalpha](std::ostream& os) -> std::ostream& {
    return hex(boolalpha(os));
  };
```
to localize the particular variety of bool and hex formatting to one place?

File: google/cloud/spanner/value_test.cc [hunk 4]
```diff
@@ -937,101 +936,124 @@ TEST(Value, CommitTimestamp) {
...
+      {MakeNullValue<Bytes>(), "NULL", normal},
+      {MakeNullValue<Date>(), "NULL", normal},
+      {MakeNullValue<Timestamp>(), "NULL", normal},
+
+      // Tests arrays
+      {Value(std::vector<bool>{false, true}), "[0, 1]", normal},
+      {Value(std::vector<bool>{false, true}), "[false, true]", boolalpha},
+      {Value(std::vector<std::int64_t>{10, 11}), "[10, 11]", normal},
+      {Value(std::vector<std::int64_t>{10, 11}), "[a, b]", hex},
+      {Value(std::vector<double>{1.0, 2.0}), "[1, 2]", normal},
```
Comment (devbww) [inline google/cloud/spanner/value_test.cc]: Add a `float4` case?

Comment (devjgm) [reply google/cloud/spanner/value_test.cc]: Done.

File: google/cloud/spanner/value_test.cc (hunk 3 above)
Comment (devjgm) [reply google/cloud/spanner/value_test.cc]: Unless you feel strongly, I think the current code is simpler and clearer and would prefer to keep it.

**Pull request merged**